base_layer/core/src/base_node/rpc/http/handler/get_header_by_height.rs

```diff
@@ -0,0 +1,40 @@
+// Copyright 2025 The Tari Project
+// SPDX-License-Identifier: BSD-3-Clause
+
+use std::sync::Arc;
+
+use axum::{extract::Query, http::StatusCode, Extension, Json};
+use log::{debug, error};
+use serde::Deserialize;
+
+use crate::{
+    base_node::rpc::{http, http::query_service::Error, BaseNodeWalletQueryService},
+    blocks::BlockHeader,
+    chain_storage::BlockchainBackend,
+};
+
+const LOG_TARGET: &str = "c::base_node::rpc::http::handler::get_header_by_height";
+
+#[derive(Deserialize)]
+pub struct QueryParams {
+    pub height: u64,
+}
+
+pub async fn handle<B: BlockchainBackend + 'static>(
```

Comment:
I think this code makes more sense in the minotari_node crate, otherwise crates that reference tari_core will include axum as a dependency

Reply:
Good point thanks! Moved to a separate crate

Reply:
Thank good point, moved to minotari_node